### PR TITLE
Support android for 'ssc list-devices'

### LINKED ISSUE #319: ssc list-devices should support Android on all build hosts
https://github.com/socketsupply/socket/blob/master/src/cli/cli.cc#L1398

### PR DESCRIPTION
**Pull request by jwerle**
Fixes #319

### CODE REVIEW

File: src/cli/templates.hh [hunk 1]
```diff
@@ -59,13 +59,13 @@ ssc v{{ssc_version}}
...
 
 options:
-  --platform  any : ios, android
-  --ecid      show device ECID
-  --udid      show device UDID
-  --only      only show ECID or UDID of the first device
+  --platform    android|ios
+  --ecid        show device ECID
+  --udid        show device UDID
+  --only        only show ECID or UDID of the first device
```
Comment (chicoxyzzy) [inline src/cli/templates.hh:68]: lets add a note that these 3 are iOS-only commands

File: npm/src/index.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ export const firstTimeExperienceSetup = async () => {
 
   let platform = ''
   process.argv.slice(2).forEach(arg => {
-    if (arg === 'setup') {
+    if (arg === 'setup' || arg === 'build') {
```
Comment (mribbons) [inline npm/src/index.js:39]: What is the reason for this change? I think it might trigger an FTE run for every build.
This isn't good if the user has elected to install android and not windows - We don't want to prompt them to install windows deps again.

File: src/cli/cli.cc [hunk 3]
```diff
@@ -1394,6 +1398,21 @@ int main (const int argc, const char* argv[]) {
         log("Could not list devices using " + command);
         exit(1);
       }
+    } else if (targetPlatform == "android") {
+      auto r = exec("adb devices | tail -n +2");
```
Comment (mribbons) [inline src/cli/cli.cc:1402]: This expects `adb` to be in `$PATH`. It looks as though you have implemented it correctly below.

**Review by mribbons — CHANGES_REQUESTED** [review]

**Review by chicoxyzzy — APPROVED** [review]

File: src/cli/cli.cc (hunk 3 above)
Comment (mribbons) [reply src/cli/cli.cc:1402]: This change could be covered in Android CI by testing for at least 1 connected emulator.

**Review by mribbons — CHANGES_REQUESTED** [review]

File: src/cli/cli.cc (hunk 3 above)
Comment (jwerle) [reply src/cli/cli.cc:1402]: oops good catch!

Comment (jwerle) [reply src/cli/cli.cc:1402]: Can come back to that type of test

**Review by mribbons — APPROVED** [review]
